p8e-api/src/main/kotlin/io/provenance/engine/config/AppConfig.kt

```diff
@@ -261,4 +264,15 @@ class AppConfig : WebMvcConfigurer {
                 }.toMap()
             MetricsService(collectors, labels)
         }
+
+    /**
+     * Add support for new key management.
+     */
+    @Bean
+    fun signer(smartKeyProperties: SmartKeyProperties, signerProperties: SignerProperties) =
```

Comment:
Since this signer is in a bean and in order to sign with it you use `setKeyId(..)` first doesn't this produce a race condition where two concurrent requests from different affiliates won't be signed correctly? Using this pattern you have I think you want this bean to return a SignerFactory.

Reply:
> I think you want this bean to return a SignerFactory.
Yea, think this what it does, in this case it's returning the "SignerImpl" class, both signer inherits from it. 

> first doesn't this produce a race condition where two concurrent requests from different affiliates won't be signed correctly?
I'd have to test this more to be absolutely sure, but its highly possible now that I think about it, personally I don't like the idea of having to `setKeyId` at the time of handling, I'd probably want the affiliateService to handle setting this to prevent any possible race condition.

Reply:
I second the idea of the bean returning a factory that can return a new instance of a SignerImpl based on what is passed in (i.e. a uuid or a keyPair) so multiple signers can operate independently

Reply:
Think I get what you guys are saying, I just need to make what I have smarter.